suite-native/discovery/src/discoveryThunks.ts

```diff
@@ -438,27 +445,13 @@ const discoverNetworkBatchThunk = createThunk(
             return;
         }
 
-        const accountType = network.accountType || NORMAL_ACCOUNT_TYPE;
-
         const lastDiscoveredAccountIndex = (round - 1) * batchSize;
 
-        // Skip Cardano legacy/ledger account types if device does not support it.
-        const isIncompatibleCardanoType =
-            network.networkType === 'cardano' &&
-            (network.accountType === 'ledger' || network.accountType === 'legacy') &&
-            !discovery.availableCardanoDerivations?.includes(network.accountType);
-
-        if (isIncompatibleCardanoType) {
-            dispatch(finishNetworkTypeDiscoveryThunk());
-
-            return;
-        }
-
```

Comment:
Already handled in `startDescriptorPreloadedDiscoveryThunk`, which is the only place which calls `discoverNetworkBatchThunk` (this function)